### ISSUE TITLE
There are too many cycles

### ISSUE THREAD

**Issue by csm-coder** [Tangshitao/Dense-Scene-Matching#8]
https://github.com/Tangshitao/Dense-Scene-Matching/blob/4957fa3f41419c31a60ffc82e234f23b8050583f/libs/engine/launcher.py#L27
should the "self.cfg.TRAIN.train_iters " be changed to "int(self.cfg.TRAIN.iters/self.cfg.TRAIN.model_save_iters)"?  otherwise，there are too many cycles.we will save 1600000 models?

**Comment by Tangshitao** [comment]
The released training codes are not guaranteed to reproduce the results in paper, but you can get very similar results.

**Comment by csm-coder** [comment]
i have trained my model basing on the best_model you provided,and also try to train a model without the best_model.but both of them can't get correct results。the errors of result are too large。the following are some loss curve and the test results in 7scenes/heads.
![图片1](https://user-images.githubusercontent.com/76400924/143882366-ce75103f-2a27-493e-9056-a4994710e888.png)
![图片2](https://user-images.githubusercontent.com/76400924/143882389-a314214c-d0b8-4c6c-b92a-f8ea9db2bc5b.png)
the model trained without best_model,batchsize=2
![图片3](https://user-images.githubusercontent.com/76400924/143882665-e8186828-dd2f-44d0-95f1-ebc0364e9ec7.png)
![图片4](https://user-images.githubusercontent.com/76400924/143882683-b18ee246-9b5e-4ed1-817a-b2a1abde8ca6.png)
the model trained without best_model,batchsize=4
![图片5](https://user-images.githubusercontent.com/76400924/143882745-5dc8358d-0411-4eaf-aad0-4bf4494ea30d.png)
![图片6](https://user-images.githubusercontent.com/76400924/143882765-1a685de2-d7f4-4dd9-97ef-915d1cf8ecbb.png)
the model trained with best_model,batchsize=8,freeze_backbone=true

i have only changed the iterations, the batchsize and whether load best_model or not.  but the results are  obviously incorrect。
I am so confused about that。

**Comment by Tangshitao** [comment]
Do you train the model on 7scenes dataset? I don't know the training curve when training the model on 7scenes dataset, but the initial loss seems to be too large when you use model_best.pth as pretrained model. It seems the model is not load correctly. Can you provide the initial training loss?

**Comment by csm-coder** [comment]
no,i train it on scannet dataset and test on 7scenes/heads. i am also confused why the initial training loss is so large. when i load the best_model.pth to test, it give correct result similar to the results in paper.this is the train loss curve when i  use model_best.pth as pretrained model.[http://115.156.215.223:8891/](url)

**Comment by Tangshitao** [comment]
I think the model does not load correctly in the training model. Can you check all parameters are loaded when training?

**Comment by csm-coder** [comment]
i have checked. the model.state_dict is loaded successfully in the training model which is the same as testing.  testing and training load the best_model in the same way. the testing result is correct, so i think the model loading should have no problem。Could there be a problem elsewhere?

**Comment by Tangshitao** [comment]
Then the problem is with data. Can you check whether the data is load correctly? For example, the unit of depth should be meters when fed into the model.

**Comment by csm-coder** [comment]
i try to train on 7scenes dataset with best_model.pth. the loss curve Looks better than it on scannet dataset. and the trained model  give a correct result.
![1](https://user-images.githubusercontent.com/76400924/144346121-d71a24c1-5e41-4cca-8c9b-ff52775572cb.png)
![2](https://user-images.githubusercontent.com/76400924/144346135-01e9d742-30d8-4010-8bb7-244ccab54f94.png)
so I consider whether there is a problem with my scannet dataset.my scannet dataset is scannet_v2,and its file path is a little bit different from your scannet label file,i have changed it. is your scannet dataset scannet_v1? and does it matter?

**Comment by Tangshitao** [comment]
I also use scannet_v2, but I also think there must be something wrong with dataloader. I cannot upload the one I use since it is too large and copyright issue. Please check this depth reading codes carefully.
